Easydict/NewApp/View/SettingView/Tabs/DisabledAppTab.swift

```diff
@@ -162,47 +169,34 @@ struct ListButton: View {
 }
 
 @available(macOS 13.0, *)
-struct BlockAppItemView: View {
-    @StateObject private var appFetcher: AppFetcher
-
+private struct BlockAppItemView: View {
     @EnvironmentObject var disabledAppViewModel: DisabledAppViewModel
 
-    init(with appModel: EZAppModel) {
-        _appFetcher = StateObject(wrappedValue: AppFetcher(appModel: appModel))
-    }
+    var appModel: EZAppModel
+
+    @State private var appIcon: NSImage = .init()
+
+    @State private var appName = ""
 
     var body: some View {
         HStack(alignment: .center) {
-            Image(nsImage: appFetcher.appIcon ?? NSImage())
+            Image(nsImage: appIcon)
                 .resizable()
                 .scaledToFit()
                 .frame(width: 24, height: 24)
 
-            Text(appFetcher.appName)
+            Text(appName)
 
             Spacer()
         }
         .frame(maxWidth: .infinity)
         .contentShape(Rectangle())
         .padding(.vertical, 4)
         .padding(.leading, 6)
-        .task {
-            appFetcher.getAppBundleInfo()
+        .onAppear {
```

Comment:
Again, I think we don't need a `onAppear` here. I'll prefer 
```swift
let appIcon: NSImage
let appName: String

init(app: EZAppModel) {
    self.appIcon = ...
    self.appName = ...
}
```

How do you think?

Reply:
`State` value could not update in `init`, so I put it in here.

If I understand right, you mean put above code in `BlockAppItemView`, but it will need decorate them as `State` or we could not change them in struct.

Reply:
Yes. I don't think we need change `appIcon` and `appName` so they don't need to be decorated with `@State`. Just put the entire logic of `getAppBundleInfo()` inside `init()`

Reply:
ok, it makes sense to me.

Reply:
Great~